File: src/achievements.py
#!/usr/bin/env python3
"""
GitHub Achievements Tracker
A simple script to track GitHub achievements
"""

class AchievementsTracker:
    def __init__(self):
        self.achievements = []
    
    def add_achievement(self, name, unlocked_date):
        """Add a new achievement"""
        self.achievements.append({
            'name': name,
            'unlocked': unlocked_date
        })
    
    def list_achievements(self):
        """List all achievements"""
        for achievement in self.achievements:
            print(f"✓ {achievement['name']} - {achievement['unlocked']}")
    
    def count(self):
        """Return total number of achievements"""
        return len(self.achievements)

if __name__ == "__main__":
    tracker = AchievementsTracker()
    print("GitHub Achievements Tracker 🏆")
    print("Track your GitHub achievements here!")

File: src/js/achievements.js
/**
 * GitHub Achievements Tracker - JavaScript Version
 * Track your GitHub achievements with this simple JavaScript module
 */

class AchievementsTracker {
    constructor() {
        this.achievements = [];
    }

    /**
     * Add a new achievement
     * @param {string} name - Achievement name
     * @param {string} unlockedDate - Date when achievement was unlocked
     */
    addAchievement(name, unlockedDate) {
        this.achievements.push({
            name,
            unlocked: unlockedDate
        });
    }

    /**
     * List all achievements
     */
    listAchievements() {
        this.achievements.forEach(achievement => {
            console.log(`✓ ${achievement.name} - ${achievement.unlocked}`);
        });
    }

    /**
     * Get total number of achievements
     * @returns {number} Total count
     */
    count() {
        return this.achievements.length;
    }
}

// Export for use in other modules
if (typeof module !== 'undefined' && module.exports) {
    module.exports = AchievementsTracker;
}
